mgmtd/mgmt_db.h

```diff
@@ -0,0 +1,432 @@
+/*
+ * MGMTD Databases
+ * Copyright (C) 2021  Vmware, Inc.
+ *		       Pushpasis Sarkar <spushpasis@vmware.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef _FRR_MGMTD_DB_H_
+#define _FRR_MGMTD_DB_H_
+
+#include "mgmtd/mgmt_defines.h"
+#include "mgmtd/mgmt_bcknd_adapter.h"
+#include "mgmtd/mgmt_frntnd_adapter.h"
+
+#define MGMTD_MAX_NUM_DBNODES_PER_BATCH 128
+
+#define MGMTD_DB_NAME_MAX_LEN 32
+#define MGMTD_DB_NAME_NONE "none"
+#define MGMTD_DB_NAME_RUNNING "running"
+#define MGMTD_DB_NAME_CANDIDATE "candidate"
+#define MGMTD_DB_NAME_OPERATIONAL "operational"
+
+#define MGMTD_STARTUP_DB_FILE_PATH "/etc/frr/frr_startup.json"
+
+#define FOREACH_MGMTD_DB_ID(id)                                                \
+	for ((id) = MGMTD_DB_NONE; (id) < MGMTD_DB_MAX_ID; (id)++)
+
+#define MGMTD_MAX_COMMIT_LIST 10
+#define MGMTD_MD5_HASH_LEN 16
+#define MGMTD_MD5_HASH_STR_HEX_LEN 33
+
+#define MGMTD_MAX_COMMIT_FILE_PATH_LEN 55
+#define MGMTD_COMMIT_FILE_PATH "/etc/frr/commit-%s.json"
+#define MGMTD_COMMIT_INDEX_FILE_NAME "/etc/frr/commit-index.dat"
+#define MGMTD_COMMIT_TIME_STR_LEN 100
+
+extern struct nb_config *running_config;
+
+struct mgmt_db_ctxt;
+
+PREDECL_DLIST(mgmt_cmt_info_dlist);
```

Comment:
The type of container should not be encoded into the name, i.e. there should not be a `_dlist` here. Same issue in multiple other places. Please remove all such suffixes.

(Rationale: the containers are explicitly designed to help in changing to another container type. If the type is encoded into the name, this breaks.)

Reply:
Sure, we will remove _dlist. Can we remove it to mgmt_cmt_info_list? That seems to me is more generic.

Reply:
I mean using _list should be okay. But not _slist or _dlist.

Reply:
`_list` is counterintuitive too — if it's e.g. changed into a RB-tree later, it will be very confusing.

I would probably pluralize the names with an `s` at the end, e.g. `mgmt_cmt_infos`.

Reply:
Ok. Sounds okay to me. Will take care of these in upcoming force push.